app/decorators/user_decorator.rb

```diff
@@ -2,7 +2,7 @@
 
 UserDecorator = Struct.new(:user) do
   def lockout_time_remaining
-    (Devise.allowed_otp_drift_seconds - (Time.zone.now - user.second_factor_locked_at)).to_i
+    (Devise.direct_otp_valid_for - (Time.zone.now - user.second_factor_locked_at)).to_i
```

Comment:
I really not a fan of locking accounts .. since it allows attackers for DOS your account pretty easily.

 But you are right, given that we are doing locking for password attempts already, we should probably do that same here...  and then remove the lockout alltogether and replace with something like captcha or exponential backoff that avoid DOS.